pkg/lib/cockpit-components-dialog.jsx

```diff
@@ -198,7 +198,7 @@ DialogFooter.propTypes = {
  * The primary action button is disabled while its action is in progress (waiting for promise)
  * Removes focus on other elements on showing
  * Expected props:
- *  - title (string)
+ *  - title (string or React component)
```

Comment:
This seems wrong? We pass the title to a modal with `<Modal title={this.props.title}>` but looking at the [PF docs](https://www.patternfly.org/components/modal/#modal) the title is a string.

Reply:
From memory, we sometimes want to have an icon in the title, and we did that by passing a non-string here. But I also think that PF now supports icons in modal titles explicitly, so we should go and fix all our code to use that.

Reply:
`<Modal titleIconVariant="warning" />`, we should change the Storage code to use that instead of putting the icon in the title "string".

Reply:
Yeah, I did check the React docs, but as the current code with passing a component works, I thought it was legit. But titleIconVariant sounds better indeed, I'll look into that!

Reply:
Done.